utils/src/main/scala/za/co/absa/enceladus/utils/schema/SchemaUtils.scala

```diff
@@ -253,8 +269,28 @@ object SchemaUtils {
     * @param fieldName Name of the field to be appended to the path
     * @return The path with the new field appended or the field itself if path is empty
     */
-  private[enceladus] def appendPath(path: String, fieldName: String) = {
-    if (path.isEmpty) fieldName else s"$path.$fieldName"
+  def appendPath(path: String, fieldName: String): String = {
+    if (path.isEmpty) {
+      fieldName
+    } else {
+      s"$path.$fieldName"
+    }
+  }
+
+  /**
+   * Combines path and field name using dot notation
+   * @param path      The dot-separated existing path
+   * @param fieldName Name of the field to be appended to the path
+   * @return          The combined path and field name
+   */
+  def fullPath(path: String, fieldName: String): String = {
```

Comment:
Method name as a verb? `appendPath()`? (not sure my suggestion is the best, I'm not super good at naming as well)

Reply:
What exactly is the difference between append path and full path? I see one more `else if` but why couldn't they be one?

Reply:
> Difference - one add adds a dot at the end of the `path` in case the `fieldName` is empty. The other not. I haven't dug that deep to verify empty string is actually a real scenario...
Main question is, why they couldn't be merged as one method. They do the same.

Reply:
OK, you convinced me. 😉 
Thinking about it, the empty `fieldName` case should not be happening and definitely not relied on.